packages/shared/src/contexts/base/BaseSharedUIProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import { pick } from 'lodash-unified'
+import { createContext, FC, useContext, useMemo, useState } from 'react'
+import type { SharedComponentOverwrite } from './types'
+
+interface ContextOptions {
+    networkIdentifier: string
+    componentOverwrite: SharedComponentOverwrite
+    updateOverwrite(overwrite: SharedComponentOverwrite): void
+    updateNetworkIdentifier(snsId: string): void
```

Comment:
```suggestion
    setOverwrite(overwrite: SharedComponentOverwrite): void
    setNetworkIdentifier(snsId: string): void
```

And SNS ID should be an enum 🤔

Reply:
Refactor in another patch? I would like to keep this one clean.